mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb

```diff
@@ -6147,6 +6145,21 @@ func containerClusterAutopilotCustomizeDiff(_ context.Context, d *schema.Resourc
 	return nil
 }
 
+// Default to VPC_NATIVE mode during initial creation
+func containerClusterNetworkingModeCustomizeDiff(_ context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	// skip if not during initial creation
+	o, _ := d.GetChange("name")
+	if o != "" {
+		return nil
+	}
+
+	err := d.SetNew("networking_mode", "VPC_NATIVE")
```

Comment:
`- .networking_mode: planned value cty.StringVal("VPC_NATIVE") does not match config value cty.StringVal("ROUTES")`

Oh, we're unconditionally setting this. We have to check the value of `n` for the field and if it's non-empty not call `SetNew`. However...

There may be a weird interaction if the value is unknown and we can't tell if it's unknown because the value is unspecified or because it's unknown based on the user's inputs, as plan will display one value and apply may eventually decide on another. It's likely safer to apply this default during apply time since user-specified unknown values will have become known. (this is messy stuff- glad to discuss over GVC if addtl details are needed, GVC mostly for expediency)

`Default` accounts for that, which CD doesn't, hence the danger of CDs. If you're familiar with "unsafe mode" as a concept, they're basically that.

Reply:
moved the logic to create and checked the value of the field first to account for this.